src/WebJobs.Script.WebHost/Metrics/HostMetricsProvider.cs

```diff
@@ -46,22 +64,42 @@ public void Start()
                 }
             };
 
+            _logger.LogInformation("Starting host metrics provider.");
+
             _meterListener.SetMeasurementEventCallback<long>(OnMeasurementRecordedLong);
             _meterListener.Start();
+
+            _started = true;
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            if (!_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                Start();
+            }
         }
 
         internal void OnMeasurementRecordedLong(
             Instrument instrument,
             long measurement,
-            ReadOnlySpan<KeyValuePair<string, object?>> tags,
-            object? state)
+            ReadOnlySpan<KeyValuePair<string, object>> tags,
+            object state)
         {
+            if (!_started)
+            {
+                return;
+            }
+
             if (instrument == null)
             {
                 throw new ArgumentNullException(nameof(instrument));
             }
 
-            AddOrUpdateMetricsCache(instrument.Name, measurement);
+            lock (_lock)
```

Comment:
Confused why some calls to AddOrUpdate are done inside a lock but others aren't. If locking is really needed, can't that be pushed down into AddOrUpdate? Or perhaps just use a concurrent dictionary like you were before?

Reply:
Yeah I think I confused everything messing with this; I think I should stick on a concurrent dictionary as before.